r/R/trajectory_analysis.R

```diff
@@ -0,0 +1,67 @@
+runTrajectoryAnalysis <- function(req, data) {
+  root_nodes <- req$body$rootNodes
+
+  cell_data <- generateGraphData(data)
+  cell_data <- monocle3::order_cells(cell_data, reduction_method = "UMAP", root_pr_nodes = root_nodes)
+
+  pseudotime <- as.data.frame(cell_data@principal_graph_aux@listData$UMAP$pseudotime)
+  pseudotime$cells_id <- data@meta.data$cells_id
+  pseudotime <- pseudotime[order(pseudotime$cells_id), ]
+  pseudotime <- pseudotime %>%
+    tidyr::complete(cells_id = seq(0, max(data@meta.data$cells_id))) %>%
```

Comment:
why is this necessary? not all cells get assigned a pseudo time?

Reply:
or is it filtered cells?

Reply:
Filtered cells